### ISSUE TITLE
Menu item with template not working correctly with Material theme

### ISSUE THREAD

**Issue by MaklaCof** [telerik/kendo-themes#484]
* Bug report

## Current behavior
If you have menu item with template which contains a html tag, navigation doesn't work.

## Expected behavior
Navigation works the same way as in default theme or menu childs.

## Minimal reproduction of the problem with instructions
[Here](https://stackblitz.com/edit/angular-mzyfr1-uwplka?file=index.html) is StackBlitz.
Mouse over primary link "My Documents" and browser doesn't navigate.
If you click any sub item (Kendo UI Projects, Reports ...) navigation is working.

## What is the motivation or use case for changing the behavior?
It's a bug. Menu navigation is not working, which is not a small thing.

## Environment
Product:
* Kendo Ui for Angular

Package versions:
Not needed StackBlitz should be enough.

Browser:
Not needed StackBlitz should be enough.

System:
Not needed StackBlitz should be enough.

**Comment by MaklaCof** [comment]
@elena-gancheva 
Do you need any additional information. When can I expect the fix. Bug is really annoying.

**Comment by MaklaCof** [comment]
I just updated to versions:
```
"@progress/kendo-theme-default": "^3.3.1",
"@progress/kendo-theme-material": "^2.3.1",
```
and problem was not fixed. After 4 months I expect this bug would be fixed.

**Comment by joneff** [comment]
He issue here is not related to the theme, but rather to your templates. You are using generic template to insert link in it. As such, the generic template is considered content and it's, quite literally , expected to see the behaviour you are seeing. If you use `kendoMenuItemLinkTemplate` and `[kendoMenuItemLink]` as [per our documentation](https://www.telerik.com/kendo-angular-ui/components/menus/menu/items/#toc-templates), you will get the expect results.

I've created an updated example: https://stackblitz.com/edit/angular-qyogca?file=app/app.component.ts.

**Comment by MaklaCof** [comment]
@joneff , @elena-gancheva 
If this is true, why the same code works with default theme? [StackBlitz](https://stackblitz.com/edit/angular-mzyfr1-lhe9qb?file=index.html)

Changed items in index.html:
```
//From
    <link rel="stylesheet" href="https://unpkg.com/@progress/kendo-theme-material@dev/dist/all.css" />

//To
    <link rel="stylesheet" href="https://unpkg.com/@progress/kendo-theme-default@dev/dist/all.css" />
```

It also works with bootstrap theme. [StackBlitz](https://stackblitz.com/edit/angular-mzyfr1-pvbsyx?file=index.html)

This points to material theme.

**Comment by MaklaCof** [comment]
This is my original html which doesn't work:
```
<kendo-menu [kendoMenuHierarchyBinding]="menus" #hierarchyMenus="kendoMenuHierarchyBinding" childrenField="childs" cssClassField="cssClasses"
            (select)="onSelect($event)">
    <ng-template kendoMenuItemTemplate let-item="item">
        <ng-container *ngIf="item.data.expression === 'Language'">
            <i class="flag-icon {{item.data.icon}}"></i>
            <span *ngIf="item.data.text">{{item.data.text}}</span>
        </ng-container>
        <ng-container *ngIf="item.data.expression !== 'Language'">
            <span *ngIf="!item.data.url">
                <fa-icon *ngIf="item.data.icon" [icon]="getIcon(item.data.icon)"></fa-icon>
                {{item.data.text}}
            </span>
            <a *ngIf="item.data.url" [routerLink]="[item.data.url]">
                <fa-icon *ngIf="item.data.icon" [icon]="getIcon(item.data.icon)"></fa-icon>
                {{item.data.text}}
            </a>
        </ng-container>
    </ng-template>
</kendo-menu>
```

I am not sure if this can be changed `kendoMenuItemLinkTemplate`. I don't have items, because I use `kendoMenuHierarchyBinding`.

**Comment by MaklaCof** [comment]
I need `a` tag, so middle button is working (open menu link in new tab).

**Comment by joneff** [comment]
I've been digging trough unresolved issues and I came across this one. And yeah, it's been more than a while since we wrote back.

Surprisingly, your examples now work and navigate. I am not sure if it is something that we did, or browser now respect empty link `<a href=""></a>`.

Either way, it's impossible to determine which one is it, due to the sheer amount of changesets both here and in the browser changelogs.
